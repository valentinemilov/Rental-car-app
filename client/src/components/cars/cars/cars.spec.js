/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';

import 'babel-polyfill';
import Cars from './cars';
import carService from '../../../services/car-service';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('should render cars data', async () => {
  const carMock = {
    id: '791c0d7e-0dda-4f72-b47c-1a4b9b66df87',
    model: 'Veyron',
    brand: 'Bugatti',
    class: 'A',
    price: 200,
    picture: 'string.jpg',
  };

  jest.spyOn(carService, 'getAllCars')
    .mockImplementation(async () => Promise.resolve([carMock]));

  await act(async () => {
    render(<Cars />, container);
  });

  expect(container.querySelector('div').className).toEqual('car-container');
});

it('should render no data', async () => {
  const carMock = {
    id: '791c0d7e-0dda-4f72-b47c-1a4b9b66df87',
    model: 'Veyron',
    brand: 'Bugatti',
    class: 'A',
    price: 200,
    picture: 'string.jpg',
  };

  jest.spyOn(carService, 'getAllCars')
    .mockImplementation(async () => Promise.resolve({
      json: () => Promise.resolve(null),
    }));

  await act(async () => {
    render(<Cars />, container);
  });

  expect(container.textContent).toBe('Loading...');
});
