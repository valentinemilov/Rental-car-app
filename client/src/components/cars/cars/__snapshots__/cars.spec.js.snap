// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly Cars component 1`] = `
<div>
  <div
    class="car-not-found"
  >
    No cars found
  </div>
</div>
`;
